src/vs/editor/contrib/inlineEdit/browser/inlineEditController.ts

```diff
@@ -0,0 +1,320 @@
+	static ID = 'editor.contrib.inlineEditController';
+
+	public static readonly inlineEditVisibleKey = 'inlineEditVisible';
+	public static readonly inlineEditVisibleContext = new RawContextKey<boolean>(InlineEditController.inlineEditVisibleKey, false);
+	private _isVisibleContext = InlineEditController.inlineEditVisibleContext.bindTo(this.contextKeyService);
+
+	public static readonly cursorAtInlineEditKey = 'cursorAtInlineEdit';
+	public static readonly cursorAtInlineEditContext = new RawContextKey<boolean>(InlineEditController.cursorAtInlineEditKey, false);
+	private _isCursorAtInlineEditContext = InlineEditController.cursorAtInlineEditContext.bindTo(this.contextKeyService);
+
+	public static get(editor: ICodeEditor): InlineEditController | null {
+		return editor.getContribution<InlineEditController>(InlineEditController.ID);
+	}
+
+	private _currentEdit: ISettableObservable<InlineEditWidget | undefined> = this._register(disposableObservableValue(this, undefined));
+	private _currentRequestCts: CancellationTokenSource | undefined;
+
+	private _jumpBackPosition: Position | undefined;
+	private _isAccepting: boolean = false;
+
+	private readonly _enabled = observableFromEvent(this.editor.onDidChangeConfiguration, () => this.editor.getOption(EditorOption.inlineEdit).enabled);
+	private readonly _fontFamily = observableFromEvent(this.editor.onDidChangeConfiguration, () => this.editor.getOption(EditorOption.inlineEdit).fontFamily);
+	private readonly _backgroundColoring = observableFromEvent(this.editor.onDidChangeConfiguration, () => this.editor.getOption(EditorOption.inlineEdit).backgroundColoring);
+
+
+	constructor(
+		public readonly editor: ICodeEditor,
+		@IInstantiationService private readonly instantiationService: IInstantiationService,
+		@IContextKeyService private readonly contextKeyService: IContextKeyService,
+		@ILanguageFeaturesService private readonly languageFeaturesService: ILanguageFeaturesService,
+		@ICommandService private readonly _commandService: ICommandService,
+		@IConfigurationService private readonly _configurationService: IConfigurationService,
+	) {
+		super();
+
+		//Automatically request inline edit when the content was changed
+		//Cancel the previous request if there is one
+		//Remove the previous ghost text
+		this._register(editor.onDidChangeModelContent(async () => {
+			if (!this._enabled.get()) {
+				return;
+			}
+			await this.getInlineEdit(editor, true);
+		}));
+
+		//Check if the cursor is at the ghost text
+		this._register(editor.onDidChangeCursorPosition((e) => {
+			if (!this._enabled.get()) {
+				return;
+			}
+			this.checkCursorPosition(e.position);
+		}));
+
+		//Perform stuff when the current edit has changed
+		this._register(autorun((reader) => {
+			/** @description InlineEditController.update model */
+			const currentEdit = this._currentEdit.read(reader);
+			if (!currentEdit) {
+				this._isVisibleContext.set(false);
+				this._isCursorAtInlineEditContext.set(false);
```

Comment:
This is always set to false, at least in both the if and else branch?

Reply:
It can be set to true in `checkCursorPosition` - but yeah, I've done some minor cleanup here.